packages/toolkit/connector-kit/src/types.ts

```diff
@@ -181,6 +182,32 @@ export type GetSession = () => Promise<ConnectorSession>;
 
 export type SetSession = (storage: ConnectorSession) => Promise<void>;
 
+export const baseStorageValueGuard = z.union([
+  z.number(),
+  z.number().array(),
+  z.string(),
+  z.string().array(),
```

Comment:
What about an array with both `number` and `string`? Or even mixed with sub array.

If this guard is too complicated, maybe we can use typescript's `JSON`